controller/metrics/clustersecretupdater.go

```diff
@@ -0,0 +1,49 @@
+package metrics
```

Comment:
I think updater does not really belong to metrics package, since it does not help to collect metrics. I would suggest to move it to `controller` package.